luigi/contrib/ftp.py

```diff
@@ -131,14 +131,12 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
-        path_parts = path.split('/')
-        path = '/'.join(path_parts[:-1])
-        fn = path_parts[-1]
+        dirname, fn = os.path.split(path)
 
-        files = self.conn.nlst(path)
+        files = self.conn.nlst(dirname)
 
         exists = False
-        if fn in files:
+        if path in files or fn in files:
```

Comment:
how about we issue `MDTM path` and `MDTM fn` (we already use MDTM below) and if both return status code 550 we decide the file doesn't exist? we might need to wrap the command in a `try` block because Python ftp module will raise if it receives a 5xx result code (see [here](https://hg.python.org/cpython/file/3.5/Lib/ftplib.py#l234)). Sorry I had no time to work on the PR.